source/common/http/filter/gzip_filter.cc

```diff
@@ -0,0 +1,159 @@
+#include "common/http/filter/gzip_filter.h"
+
+#include <regex>
+
+#include "common/common/macros.h"
+
+namespace Envoy {
+namespace Http {
+
+static const std::regex& acceptEncodingRegex() {
+  CONSTRUCT_ON_FIRST_USE(std::regex, "(?!.*gzip;\\s*q=0(,|$))(?=(.*gzip)|(^\\*$))",
+                         std::regex::optimize);
+}
+
+GzipFilterConfig::GzipFilterConfig(const envoy::api::v2::filter::http::Gzip& gzip)
+    : compression_level_(compressionLevelEnum(gzip.compression_level())),
+      compression_strategy_(compressionStrategyEnum(gzip.compression_strategy())),
+      content_length_(static_cast<uint64_t>(gzip.content_length().value())),
+      memory_level_(static_cast<uint64_t>(gzip.memory_level().value())),
+      etag_(gzip.disable_on_etag().value()),
+      last_modified_(gzip.disable_on_last_modified().value()) {
+
+  for (const auto& value : gzip.cache_control()) {
+    cache_control_values_.insert(value);
+  }
```

Comment:
I think there's a simpler way to initialize these which will eliminate the constructor body and allow you to make all the members of the `GzipFilterConfig` const.  See the part about RepeatedPtrField in https://developers.google.com/protocol-buffers/docs/reference/cpp-generated

This should work: `cache_control_values_(gzip.cache_control().cbegin(), gzip.cache_control.cend())`

Reply:
I knew that repeated provides STL-like iterators, but I didn't realized that it could be initialized like that. Cool, thanks! 👍